Sources/Apodini/Semantic Model Builder/gRPC/GRPCInterfaceExporter.swift

```diff
@@ -36,105 +36,91 @@ class GRPCInterfaceExporter: InterfaceExporter {
                 self.parameters[param.element.id] = param.offset + 1
             }
 
-        let context = endpoint.createConnectionContext(for: self)
-
         // expose the new component via a GRPCService
         // currently unary enpoints are considered here
-        if let service = services[serviceName] {
-            service.exposeUnaryEndpoint(name: methodName, context: context)
+        let service: GRPCService
+        if let srvc = services[serviceName] {
+            service = srvc
         } else {
-            let service = GRPCService(name: serviceName, using: app)
-            service.exposeUnaryEndpoint(name: methodName, context: context)
+            service = GRPCService(name: serviceName, using: app)
             services[serviceName] = service
         }
-        
-        app.logger.info("Exported gRPC endpoint \(serviceName)/\(methodName) with parameters:")
-        for parameter in endpoint.parameters {
-            app.logger.info("\t\(parameter.propertyType) \(parameter.name) = \(getFieldTag(for: parameter) ?? 0);")
-        }
-    }
 
-    /// The GRPC exporter handles all parameters equally as body parameters
-    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: Vapor.Request) throws -> Type?? {
-        let contentType = request.headers.first(name: .contentType)
-        switch contentType {
-        case "application/grpc", "application/grpc+proto":
-            guard let fieldTag = getFieldTag(for: parameter) else {
-                // If this occurs, something went fundamentally wrong in usage
-                // of the GRPC exporter.
-                // Each parameter should get a default field tag assigned
-                // above in the export() funtction.
-                fatalError("No default or explicit field tag available")
-            }
-            guard let data = bodyData(from: request) else {
-                throw GRPCError.decodingError("""
-                    No body data available to decode from.
-                    GRPC exporter expects all parameters to be in the body of the request.
+        let context = endpoint.createConnectionContext(for: self)
+
+        do {
+            if endpoint.serviceType == .unary {
+                try service.exposeUnaryEndpoint(name: methodName, context: context)
+                app.logger.info("Exported unary gRPC endpoint \(serviceName)/\(methodName)")
+            } else if endpoint.serviceType == .clientStreaming {
+                try service.exposeClientStreamingEndpoint(name: methodName, context: context)
+                app.logger.info("Exported client-streaming gRPC endpoint \(serviceName)/\(methodName)")
+            } else {
+                app.logger.warning("""
```

Comment:
"#warnings  should be moved to more descriptive and trackable GitHub issues that can be addressed step by step" 😄 this could be moved to an issue, couldn't it?

Reply:
You're right. I opened issue #142 and put a reference in the code.

However, I would still like to keep the warning-log for the user's information (as it's not a `#warning` but merely a log for the Apodini-user to understand what's happening).